libV2/utils/timestamp.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * Converts the given UNIX timestamp to microsecond precision
+ * @param {Number} timestamp - timestamp to pad
+ * @returns {Number} - Padded timestamp
+ */
+function convertTimestamp(timestamp) {
+    if (timestamp < 10000000000) { // Second precision
```

Comment:
This way of doing things looks like a hack to me, even though in practice it should work, does not look very clean. Is there a way to split this function in two different helpers for seconds/milliseconds cases, and call the appropriate helper each time? Or do the math directly instead of calling a helper since it looks like it's a simple multiplication that may not require a helper function.

Reply:
My reason for abstracting this is that is requires usage of several "magic" numbers. Initially my thoughts were to add constants and do the logic in the places it was needed, but decided that since a wrong conversion could lead to subtle bugs (I've already made this mistake personally several times) that adding a util function made sense. With regards to the use of `<` I can't think of a different way to distinguish timestamps of a particular precision

Reply:
OK. I still think keeping the timestamp type around (milli or micro, essentially) would be cleaner than guessing based on the value range, but understand your point, not blocking just may be worth rethinking in the future.

Reply:
One of the uses for this is the timestamp that is passed into the API which does not come with an attached precision. Currently we only support seconds, but I wanted to lift this as the underlying platform has better resolution now. Warp10 neatly sidesteps this problem by requiring timestamps to be in ISO8601 format so that the precision can be recovered server side while allowing users to use any precision they like. Might be a nice change for API v2

Reply:
Moment.js has a helper for this https://momentjs.com/docs/#/parsing/unix-timestamp/